src/Functions/UTCTimestampTransform.cpp

```diff
@@ -0,0 +1,184 @@
+
+        static FunctionPtr create(ContextPtr) { return std::make_shared<UTCTimestampTransform>(); }
+
+        DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+        {
+            if (arguments.size() < 2)
+                throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH, "Function {}'s arguments number must be 2.", name);
+
+            DataTypePtr first_arg_type = arguments[0];
+            DataTypePtr second_arg_type = arguments[1];
+            if (first_arg_type->isNullable())
+            {
+                const DataTypeNullable * null_type = static_cast<const DataTypeNullable *>(first_arg_type.get());
+                first_arg_type = null_type->getNestedType();
+            }
+            if (second_arg_type->isNullable())
+            {
+                const DataTypeNullable * null_type = static_cast<const DataTypeNullable *>(second_arg_type.get());
+                second_arg_type = null_type->getNestedType();
+            }
+            WhichDataType which_type_first(arguments[0]);
+            WhichDataType which_type_second(arguments[1]);
+            if (!which_type_first.isDateTime() && !which_type_first.isDateTime64())
+                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Function {}'s 1st argument type must be datetime.", name);
+            if (!which_type_second.isString())
+                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Function {}'s 2nd argument type must be string.", name);
+            DataTypePtr date_time_type;
+            if (which_type_first.isDateTime())
+                date_time_type = std::make_shared<DataTypeDateTime>();
+            else
+            {
+                const DataTypeDateTime64 * date_time_64 = static_cast<const DataTypeDateTime64 *>(first_arg_type.get());
+                date_time_type = std::make_shared<DataTypeDateTime64>(date_time_64->getScale());
+            }
+            return std::make_shared<DataTypeNullable>(date_time_type);
+        }
+
+        ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t input_rows_count) const override
+        {
+            if (arguments.size() < 2)
+                throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH, "Function {}'s arguments number must be 2.", name);
+            ColumnWithTypeAndName arg1 = arguments[0];
+            ColumnWithTypeAndName arg2 = arguments[1];
+            if (arg1.column->size() != input_rows_count || arg2.column->size() != input_rows_count)
+                throw Exception(
+                    ErrorCodes::ILLEGAL_COLUMN,
+                    "The argument column size {} is not match input rows count {}.",
+                    arg1.column->size() != input_rows_count ? arg1.column->size() : arg2.column->size(),
+                    input_rows_count);
```

Comment:
Why? If I remember correctly, `input_rows_count` means that we should process first `input_rows_count` rows of provided columns and it's guaranteed that their sizes not less then `input_rows_count`.